src/core/context.rs

```diff
@@ -1,52 +1,65 @@
 use std::marker::PhantomData;
 use std::sync::Arc;
 
-use crate::core::types::{precomputed_square, RcVar, Type, Variable, VariableFactory};
 use crate::jet::Jet;
+use crate::types;
 
 /// Context for constructing a Simplicity program
 #[allow(dead_code)]
 pub struct Context<J: Jet> {
-    pub(crate) naming: VariableFactory,
-    pow2: Vec<RcVar>,
+    pub(crate) naming: types::variable::Factory,
+    pow22: Vec<types::Type>,
```

Comment:
Should be `pow2`. Remnant of when there used to exist `pow2: Vec<RcVar>`.

Reply:
I can rename to `pow2` (it was originally `pow22` to avoid conflicting with the existing `pow2` field, which later gets deleted). But I think `pow22` is actually a bit better, because these are types of the form `2^(2^n)`.

I will add some more comments to these types because this has caused me a lot of confusion.